pydicom/tests/test_rle_pixel_data.py

```diff
@@ -1,126 +1,762 @@
+        assert ds.SamplesPerPixel == 3
+        assert ds.NumberOfFrames == 2
+        arr = ds.pixel_array
+        ref = _get_pixel_array(SC_EXPL_LITTLE_16_2F)
+
+        assert np.array_equal(ds.pixel_array, ref)
+
+        # Frame 1
+        frame = arr[0]
+        assert (65535, 0, 0) == tuple(frame[5, 50, :])
+        assert (65535, 32896, 32896) == tuple(frame[15, 50, :])
+        assert (0, 65535, 0) == tuple(frame[25, 50, :])
+        assert (32896, 65535, 32896) == tuple(frame[35, 50, :])
+        assert (0, 0, 65535) == tuple(frame[45, 50, :])
+        assert (32896, 32896, 65535) == tuple(frame[55, 50, :])
+        assert (0, 0, 0) == tuple(frame[65, 50, :])
+        assert (16448, 16448, 16448) == tuple(frame[75, 50, :])
+        assert (49344, 49344, 49344) == tuple(frame[85, 50, :])
+        assert (65535, 65535, 65535) == tuple(frame[95, 50, :])
+
+        # Frame 2 is frame 1 inverted
+        assert np.array_equal((2**ds.BitsAllocated - 1) - arr[1], arr[0])
+
+    def test_pixel_array_32bit_1sample_1f(self):
+        """Test pixel_array for 32-bit, 1 sample/pixel, 1 frame."""
+        ds = dcmread(RTDOSE_RLE_1F)
+        assert ds.file_meta.TransferSyntaxUID == RLELossless
+        assert ds.BitsAllocated == 32
+        assert ds.SamplesPerPixel == 1
+        assert 'NumberOfFrames' not in ds
+        ref = _get_pixel_array(RTDOSE_EXPL_LITTLE_1F)
+        arr = ds.pixel_array
+
+        assert np.array_equal(arr, ref)
+        assert (10, 10) == arr.shape
+        assert (1249000, 1249000, 1250000) == tuple(arr[0, :3])
+        assert (1031000, 1029000, 1027000) == tuple(arr[4, 3:6])
+        assert (803000, 801000, 798000) == tuple(arr[-1, -3:])
+
+    def test_pixel_array_32bit_1sample_15f(self):
+        """Test pixel_array for 32-bit, 1, sample/pixel, 15 frame."""
+        ds = dcmread(RTDOSE_RLE_15F)
+        assert ds.file_meta.TransferSyntaxUID == RLELossless
+        assert ds.BitsAllocated == 32
+        assert ds.SamplesPerPixel == 1
+        assert ds.NumberOfFrames == 15
+        ref = _get_pixel_array(RTDOSE_EXPL_LITTLE_15F)
+        arr = ds.pixel_array
+
+        assert np.array_equal(arr, ref)
+        assert (15, 10, 10) == arr.shape
+
+        # Frame 1
+        assert (1249000, 1249000, 1250000) == tuple(arr[0, 0, :3])
+        assert (1031000, 1029000, 1027000) == tuple(arr[0, 4, 3:6])
+        assert (803000, 801000, 798000) == tuple(arr[0, -1, -3:])
+
+        # Frame 8
+        assert (1253000, 1253000, 1249000) == tuple(arr[7, 0, :3])
+        assert (1026000, 1023000, 1022000) == tuple(arr[7, 4, 3:6])
+        assert (803000, 803000, 803000) == tuple(arr[7, -1, -3:])
+
+        # Frame 15
+        assert (1249000, 1250000, 1251000) == tuple(arr[-1, 0, :3])
+        assert (1031000, 1031000, 1031000) == tuple(arr[-1, 4, 3:6])
+        assert (801000, 800000, 799000) == tuple(arr[-1, -1, -3:])
+
+    @pytest.mark.skip(reason='Samples/pixel>1, BitsAllocated>8 not supported')
```

Comment:
Is that not supported only if Number Of Frames is 1? The next test seems to suggest that... (same for 16 bits allocated). Where does this limitation come from?

Reply:
`Dataset._reshape_pixel_array` raises `NotImplemented` if frames = 1, samples per pixel > 1 and bits allocated > 8. Not sure why it was originally added, perhaps because of the incomplete support for Planar Configuration?

#720 fixes this.